src/snowflake/cli/_plugins/stage/manager.py

```diff
@@ -132,9 +133,14 @@ class DefaultStagePathParts(StagePathParts):
     """
 
     def __init__(self, stage_path: str):
-        self.directory = self.get_directory(stage_path)
-        self.stage = StageManager.get_stage_from_path(stage_path)
-        stage_name = self.stage.split(".")[-1]
+        match = re.fullmatch(STAGE_PATH_REGEX, stage_path)
+        if match is None:
+            raise ClickException("Invalid stage path")
```

Comment:
this validation should be done in Pydantic model. For users, this might not be clear where this exception is coming from.

Reply:
we have validations in the pydantic model for NA. this is for other uses of this stage object.